In [1]:
import pandas as pd

In [2]:
outcomes = pd.read_csv('aac_outcomes.csv')

In [3]:
intakes = pd.read_csv('aac_intakes.csv')

In [4]:
intake_outcomes = pd.read_csv('aac_intakes_outcomes.csv')

In [5]:
intake_outcomes.dtypes

age_upon_outcome               object
animal_id_outcome              object
date_of_birth                  object
outcome_subtype                object
outcome_type                   object
sex_upon_outcome               object
age_upon_outcome_(days)         int64
age_upon_outcome_(years)      float64
age_upon_outcome_age_group     object
outcome_datetime               object
outcome_month                   int64
outcome_year                    int64
outcome_monthyear              object
outcome_weekday                object
outcome_hour                    int64
outcome_number                float64
dob_year                        int64
dob_month                       int64
dob_monthyear                  object
age_upon_intake                object
animal_id_intake               object
animal_type                    object
breed                          object
color                          object
found_location                 object
intake_condition               object
intake_type 

In [6]:
intake_outcomes['outcome_datetime'] = pd.to_datetime(intake_outcomes['outcome_datetime'], format='mixed')


In [7]:
intake_outcomes['intake_datetime'] = pd.to_datetime(intake_outcomes['intake_datetime'], format='mixed')


In [8]:
intake_outcomes['sex_upon_intake'] = intake_outcomes['sex_upon_intake'].str.strip().str.lower()
intake_outcomes['sex_upon_outcome'] = intake_outcomes['sex_upon_outcome'].str.strip().str.lower()

In [9]:
intakes.dtypes

age_upon_intake     object
animal_id           object
animal_type         object
breed               object
color               object
datetime            object
datetime2           object
found_location      object
intake_condition    object
intake_type         object
name                object
sex_upon_intake     object
dtype: object

In [10]:
outcomes.dtypes

age_upon_outcome    object
animal_id           object
animal_type         object
breed               object
color               object
date_of_birth       object
datetime            object
monthyear           object
name                object
outcome_subtype     object
outcome_type        object
sex_upon_outcome    object
dtype: object

In [11]:
outcomes['datetime'] = pd.to_datetime(outcomes['datetime'], format='mixed')


In [12]:
outcomes.dtypes

age_upon_outcome            object
animal_id                   object
animal_type                 object
breed                       object
color                       object
date_of_birth               object
datetime            datetime64[ns]
monthyear                   object
name                        object
outcome_subtype             object
outcome_type                object
sex_upon_outcome            object
dtype: object

In [13]:
intake_outcomes.head()
intake_outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79672 entries, 0 to 79671
Data columns (total 41 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   age_upon_outcome            79672 non-null  object        
 1   animal_id_outcome           79672 non-null  object        
 2   date_of_birth               79672 non-null  object        
 3   outcome_subtype             36348 non-null  object        
 4   outcome_type                79662 non-null  object        
 5   sex_upon_outcome            79671 non-null  object        
 6   age_upon_outcome_(days)     79672 non-null  int64         
 7   age_upon_outcome_(years)    79672 non-null  float64       
 8   age_upon_outcome_age_group  79672 non-null  object        
 9   outcome_datetime            79672 non-null  datetime64[ns]
 10  outcome_month               79672 non-null  int64         
 11  outcome_year                79672 non-null  int64     

In [14]:
intake_outcomes['outcome_type'] = intake_outcomes['outcome_type'].str.strip().str.lower()

In [15]:
intakes.head()

age_upon_intake animal_id animal_type  \
0         8 years   A706918         Dog   
1       11 months   A724273         Dog   
2         4 weeks   A665644         Cat   
3         4 years   A682524         Dog   
4         2 years   A743852         Dog   

                                   breed        color  \
0               English Springer Spaniel  White/Liver   
1                            Basenji Mix  Sable/White   
2                 Domestic Shorthair Mix       Calico   
3  Doberman Pinsch/Australian Cattle Dog     Tan/Gray   
4                 Labrador Retriever Mix    Chocolate   

                  datetime                datetime2  \
0  2015-07-05T12:59:00.000  2015-07-05T12:59:00.000   
1  2016-04-14T18:43:00.000  2016-04-14T18:43:00.000   
2  2013-10-21T07:59:00.000  2013-10-21T07:59:00.000   
3  2014-06-29T10:38:00.000  2014-06-29T10:38:00.000   
4  2017-02-18T12:46:00.000  2017-02-18T12:46:00.000   

                       found_location intake_condition      intake_type  \
0    9409 Bluegrass Dr in Austin (TX)           Normal            Stray   
1  2818 Palomino Trail in Austin (TX)           Normal            Stray   
2                         Austin (TX)             Sick            Stray   
3       800 Grove Blvd in Austin (TX)           Normal            Stray   
4                         Austin (TX)           Normal  Owner Surrender   

      name sex_upon_intake  
0    Belle   Spayed Female  
1  Runster     Intact Male  
2      NaN   Intact Female  
3      Rio   Neutered Male  
4     Odin   Neutered Male

In [16]:
outcomes.head()

age_upon_outcome animal_id animal_type                    breed  \
0          2 weeks   A684346         Cat   Domestic Shorthair Mix   
1           1 year   A666430         Dog               Beagle Mix   
2           1 year   A675708         Dog                 Pit Bull   
3          9 years   A680386         Dog  Miniature Schnauzer Mix   
4         5 months   A683115       Other                  Bat Mix   

          color        date_of_birth            datetime            monthyear  \
0  Orange Tabby  2014-07-07T00:00:00 2014-07-22 16:04:00  2014-07-22T16:04:00   
1   White/Brown  2012-11-06T00:00:00 2013-11-07 11:47:00  2013-11-07T11:47:00   
2    Blue/White  2013-03-31T00:00:00 2014-06-03 14:20:00  2014-06-03T14:20:00   
3         White  2005-06-02T00:00:00 2014-06-15 15:50:00  2014-06-15T15:50:00   
4         Brown  2014-01-07T00:00:00 2014-07-07 14:04:00  2014-07-07T14:04:00   

      name outcome_subtype outcome_type sex_upon_outcome  
0      NaN         Partner     Transfer      Intact Male  
1     Lucy         Partner     Transfer    Spayed Female  
2  *Johnny             NaN     Adoption    Neutered Male  
3   Monday         Partner     Transfer    Neutered Male  
4      NaN     Rabies Risk   Euthanasia          Unknown

In [17]:
outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80681 entries, 0 to 80680
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   age_upon_outcome  80673 non-null  object        
 1   animal_id         80681 non-null  object        
 2   animal_type       80681 non-null  object        
 3   breed             80681 non-null  object        
 4   color             80681 non-null  object        
 5   date_of_birth     80681 non-null  object        
 6   datetime          80681 non-null  datetime64[ns]
 7   monthyear         80681 non-null  object        
 8   name              56116 non-null  object        
 9   outcome_subtype   36893 non-null  object        
 10  outcome_type      80667 non-null  object        
 11  sex_upon_outcome  80679 non-null  object        
dtypes: datetime64[ns](1), object(11)
memory usage: 7.4+ MB


In [18]:
intakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80187 entries, 0 to 80186
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   age_upon_intake   80187 non-null  object
 1   animal_id         80187 non-null  object
 2   animal_type       80187 non-null  object
 3   breed             80187 non-null  object
 4   color             80187 non-null  object
 5   datetime          80187 non-null  object
 6   datetime2         80187 non-null  object
 7   found_location    80187 non-null  object
 8   intake_condition  80187 non-null  object
 9   intake_type       80187 non-null  object
 10  name              55603 non-null  object
 11  sex_upon_intake   80186 non-null  object
dtypes: object(12)
memory usage: 7.3+ MB


In [19]:
combined_df = pd.merge(
    intakes,
    outcomes,
    on =  'animal_id',
    how = 'left',
    suffixes=('_intakes', '_outcomes')
)

In [20]:
combined_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100627 entries, 0 to 100626
Data columns (total 23 columns):
 #   Column                Non-Null Count   Dtype         
---  ------                --------------   -----         
 0   age_upon_intake       100627 non-null  object        
 1   animal_id             100627 non-null  object        
 2   animal_type_intakes   100627 non-null  object        
 3   breed_intakes         100627 non-null  object        
 4   color_intakes         100627 non-null  object        
 5   datetime_intakes      100627 non-null  object        
 6   datetime2             100627 non-null  object        
 7   found_location        100627 non-null  object        
 8   intake_condition      100627 non-null  object        
 9   intake_type           100627 non-null  object        
 10  name_intakes          75527 non-null   object        
 11  sex_upon_intake       100626 non-null  object        
 12  age_upon_outcome      100226 non-null  object        
 13 

## 1)Find the top 5 places where animals are found so the shelter can coordinate with local volunteers and animal control to monitor these areas.

## *Areas where more animals are found

In [21]:
top_5_places = intakes['found_location'].value_counts().head(5)
print(top_5_places)

found_location
Austin (TX)                          14443
Outside Jurisdiction                   948
Travis (TX)                            921
7201 Levander Loop in Austin (TX)      517
Del Valle (TX)                         411
Name: count, dtype: int64


### *Top 5 places found by counting the number of occurences based off the "found_location" column

## 2)What is the average number of pets found in a month in the year 2015? Are there months where there is a higher number of animals found?


In [22]:
intakes.info()
intakes = intakes.drop(columns=['datetime2'])
intakes['datetime'] = pd.to_datetime(intakes['datetime'])

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80187 entries, 0 to 80186
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   age_upon_intake   80187 non-null  object
 1   animal_id         80187 non-null  object
 2   animal_type       80187 non-null  object
 3   breed             80187 non-null  object
 4   color             80187 non-null  object
 5   datetime          80187 non-null  object
 6   datetime2         80187 non-null  object
 7   found_location    80187 non-null  object
 8   intake_condition  80187 non-null  object
 9   intake_type       80187 non-null  object
 10  name              55603 non-null  object
 11  sex_upon_intake   80186 non-null  object
dtypes: object(12)
memory usage: 7.3+ MB


In [23]:
intakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80187 entries, 0 to 80186
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   age_upon_intake   80187 non-null  object        
 1   animal_id         80187 non-null  object        
 2   animal_type       80187 non-null  object        
 3   breed             80187 non-null  object        
 4   color             80187 non-null  object        
 5   datetime          80187 non-null  datetime64[ns]
 6   found_location    80187 non-null  object        
 7   intake_condition  80187 non-null  object        
 8   intake_type       80187 non-null  object        
 9   name              55603 non-null  object        
 10  sex_upon_intake   80186 non-null  object        
dtypes: datetime64[ns](1), object(10)
memory usage: 6.7+ MB


In [24]:
intakes_2015 = intakes[intakes['datetime'].dt.year == 2015]

In [25]:
intakes_2015.head()

age_upon_intake animal_id animal_type                     breed  \
0          8 years   A706918         Dog  English Springer Spaniel   
5          2 years   A708452         Dog    Labrador Retriever Mix   
8         5 months   A707375         Dog                  Pit Bull   
9          2 years   A696408         Dog       Chihuahua Shorthair   
12         4 weeks   A707658         Cat    Domestic Shorthair Mix   

          color            datetime  \
0   White/Liver 2015-07-05 12:59:00   
5   Black/White 2015-07-30 14:37:00   
8   Brown/White 2015-07-11 18:19:00   
9      Tricolor 2015-02-04 12:58:00   
12  Black/White 2015-07-15 17:43:00   

                                  found_location intake_condition  \
0               9409 Bluegrass Dr in Austin (TX)           Normal   
5                                    Austin (TX)           Normal   
8   Galilee Court And Damita Jo Dr in Manor (TX)           Normal   
9                    9705 Thaxton in Austin (TX)           Normal   
12                   6118 Fairway in Austin (TX)           Normal   

      intake_type         name sex_upon_intake  
0           Stray        Belle   Spayed Female  
5   Public Assist       Mumble     Intact Male  
8           Stray  *Candy Cane   Intact Female  
9           Stray       *Pearl   Intact Female  
12          Stray        *Mint   Intact Female

In [26]:
count_by_month = intakes_2015.groupby(intakes_2015['datetime'].dt.month).size()

In [27]:
count_by_month

datetime
1     1198
2     1119
3     1346
4     1543
5     2094
6     2189
7     1635
8     1718
9     1591
10    1740
11    1411
12    1128
dtype: int64

In [28]:
count_by_month.info()

<class 'pandas.core.series.Series'>
Index: 12 entries, 1 to 12
Series name: None
Non-Null Count  Dtype
--------------  -----
12 non-null     int64
dtypes: int64(1)
memory usage: 144.0 bytes


In [29]:
count_by_month.head(5)

datetime
1    1198
2    1119
3    1346
4    1543
5    2094
dtype: int64

In [30]:
monthly_avg = count_by_month.mean()

## ** Monthly Average for 2015

In [31]:
print(f'Monthly average for pets found in 2015 is {round(monthly_avg)}')

Monthly average for pets found in 2015 is 1559


In [32]:
top_5_months = count_by_month.sort_values(ascending=False).head(5)

In [33]:
top_5_months

datetime
6     2189
5     2094
10    1740
8     1718
7     1635
dtype: int64

## **It seems like the summer months and October are the months when more animals are found

### **To find this conclusion, we first convert datetime to a date object. Then we get the year of 2015. Then calculate by month. When we sort by the count in the month, we can pull the top 5 by using the sort_values and head() functions

## 3) What is the ratio of incoming pets vs. adopted pets?

In [34]:
intakes.info()
outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80187 entries, 0 to 80186
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   age_upon_intake   80187 non-null  object        
 1   animal_id         80187 non-null  object        
 2   animal_type       80187 non-null  object        
 3   breed             80187 non-null  object        
 4   color             80187 non-null  object        
 5   datetime          80187 non-null  datetime64[ns]
 6   found_location    80187 non-null  object        
 7   intake_condition  80187 non-null  object        
 8   intake_type       80187 non-null  object        
 9   name              55603 non-null  object        
 10  sex_upon_intake   80186 non-null  object        
dtypes: datetime64[ns](1), object(10)
memory usage: 6.7+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80681 entries, 0 to 80680
Data columns (total 12 columns):
 #   Column         

In [35]:
intake_outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79672 entries, 0 to 79671
Data columns (total 41 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   age_upon_outcome            79672 non-null  object        
 1   animal_id_outcome           79672 non-null  object        
 2   date_of_birth               79672 non-null  object        
 3   outcome_subtype             36348 non-null  object        
 4   outcome_type                79662 non-null  object        
 5   sex_upon_outcome            79671 non-null  object        
 6   age_upon_outcome_(days)     79672 non-null  int64         
 7   age_upon_outcome_(years)    79672 non-null  float64       
 8   age_upon_outcome_age_group  79672 non-null  object        
 9   outcome_datetime            79672 non-null  datetime64[ns]
 10  outcome_month               79672 non-null  int64         
 11  outcome_year                79672 non-null  int64     

In [36]:
intakes_no_dup = intakes.drop_duplicates()

In [37]:
intakes_no_dup.info()

<class 'pandas.core.frame.DataFrame'>
Index: 80161 entries, 0 to 80186
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   age_upon_intake   80161 non-null  object        
 1   animal_id         80161 non-null  object        
 2   animal_type       80161 non-null  object        
 3   breed             80161 non-null  object        
 4   color             80161 non-null  object        
 5   datetime          80161 non-null  datetime64[ns]
 6   found_location    80161 non-null  object        
 7   intake_condition  80161 non-null  object        
 8   intake_type       80161 non-null  object        
 9   name              55586 non-null  object        
 10  sex_upon_intake   80160 non-null  object        
dtypes: datetime64[ns](1), object(10)
memory usage: 7.3+ MB


In [38]:
outcomes_no_dup = outcomes.drop_duplicates()

In [39]:
outcomes_no_dup.info()

<class 'pandas.core.frame.DataFrame'>
Index: 80671 entries, 0 to 80680
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   age_upon_outcome  80663 non-null  object        
 1   animal_id         80671 non-null  object        
 2   animal_type       80671 non-null  object        
 3   breed             80671 non-null  object        
 4   color             80671 non-null  object        
 5   date_of_birth     80671 non-null  object        
 6   datetime          80671 non-null  datetime64[ns]
 7   monthyear         80671 non-null  object        
 8   name              56111 non-null  object        
 9   outcome_subtype   36888 non-null  object        
 10  outcome_type      80657 non-null  object        
 11  sex_upon_outcome  80669 non-null  object        
dtypes: datetime64[ns](1), object(11)
memory usage: 8.0+ MB


In [40]:
ratio2 = len(intakes_no_dup) / (outcomes_no_dup["outcome_type"] == "Adoption").sum()

In [41]:
#sort so the adopted outcome comes first 
#and not dropped from duplicates
cleaned_combo = intake_outcomes.sort_values(
    by=["animal_id_outcome", "outcome_type"], ascending=[True,True])

In [42]:
cleaned_combo.info()

<class 'pandas.core.frame.DataFrame'>
Index: 79672 entries, 0 to 79671
Data columns (total 41 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   age_upon_outcome            79672 non-null  object        
 1   animal_id_outcome           79672 non-null  object        
 2   date_of_birth               79672 non-null  object        
 3   outcome_subtype             36348 non-null  object        
 4   outcome_type                79662 non-null  object        
 5   sex_upon_outcome            79671 non-null  object        
 6   age_upon_outcome_(days)     79672 non-null  int64         
 7   age_upon_outcome_(years)    79672 non-null  float64       
 8   age_upon_outcome_age_group  79672 non-null  object        
 9   outcome_datetime            79672 non-null  datetime64[ns]
 10  outcome_month               79672 non-null  int64         
 11  outcome_year                79672 non-null  int64         


In [43]:
ratio = len(cleaned_combo) / (cleaned_combo["outcome_type"] == "adoption").sum()

In [44]:
ratio

2.3716139786866703

## ***The ratio calculated from number of pets incoming to adopted pets is 2.37

In [45]:
###########################################

## 4) What is the distribution of the types of animals in the shelter?
## Find the count of each type of animal in the shelter.

In [46]:
animal_counts = cleaned_combo["animal_type"].value_counts()

In [47]:
print(animal_counts)

animal_type
Dog      45366
Cat      29539
Other     4428
Bird       339
Name: count, dtype: int64


## ****Distribution of animals 
Dogs = 45366
Cats = 29539
Birds = 339
Other = 4428

## ----------------------------------------

## 5)What are the adoption rates for specific breeds?
## Find the top 5 dog breeds in the shelter (based on count) and then find the adoption percentage of each breed

In [48]:
dogs = cleaned_combo[cleaned_combo["animal_type"] == "Dog"]

In [49]:
print(dogs.shape)

(45366, 41)


In [50]:
print(dogs.head())

  age_upon_outcome animal_id_outcome        date_of_birth outcome_subtype  \
0         10 years           A006100  2007-07-09 00:00:00             NaN   
1          7 years           A006100  2007-07-09 00:00:00             NaN   
2          6 years           A006100  2007-07-09 00:00:00             NaN   
3         10 years           A047759  2004-04-02 00:00:00         Partner   
4         16 years           A134067  1997-10-16 00:00:00             NaN   

      outcome_type sex_upon_outcome  age_upon_outcome_(days)  \
0  return to owner    neutered male                     3650   
1  return to owner    neutered male                     2555   
2  return to owner    neutered male                     2190   
3         transfer    neutered male                     3650   
4  return to owner    neutered male                     5840   

   age_upon_outcome_(years) age_upon_outcome_age_group    outcome_datetime  \
0                      10.0                (7.5, 10.0] 2017-12-07 14:07:00

In [51]:
dogs.dtypes

age_upon_outcome                      object
animal_id_outcome                     object
date_of_birth                         object
outcome_subtype                       object
outcome_type                          object
sex_upon_outcome                      object
age_upon_outcome_(days)                int64
age_upon_outcome_(years)             float64
age_upon_outcome_age_group            object
outcome_datetime              datetime64[ns]
outcome_month                          int64
outcome_year                           int64
outcome_monthyear                     object
outcome_weekday                       object
outcome_hour                           int64
outcome_number                       float64
dob_year                               int64
dob_month                              int64
dob_monthyear                         object
age_upon_intake                       object
animal_id_intake                      object
animal_type                           object
breed     

In [52]:
breed_counts = dogs['breed'].value_counts()

In [53]:
top5_breeds = breed_counts.head(5)

In [54]:
print(top5_breeds)

breed
Pit Bull Mix                 6256
Chihuahua Shorthair Mix      4831
Labrador Retriever Mix       4789
German Shepherd Mix          1950
Australian Cattle Dog Mix    1099
Name: count, dtype: int64


In [55]:
percentage_adopted = {}
for breed in top5_breeds.index:
    total = dogs[dogs['breed'] == breed].shape[0]
    adopted = dogs[(dogs['breed'] == breed) & (dogs['outcome_type'] =='adoption')].shape[0]
    percentage_adopted[breed] = (adopted / total) * 100

In [56]:
print(f'{total} {adopted} {percentage_adopted}')

1099 607 {'Pit Bull Mix': 37.148337595907925, 'Chihuahua Shorthair Mix': 46.98820120057959, 'Labrador Retriever Mix': 49.446648569638754, 'German Shepherd Mix': 47.28205128205128, 'Australian Cattle Dog Mix': 55.23202911737943}


## *****Adoption rates based on breed

In [57]:
for breed, pct in percentage_adopted.items():
    print(f"{breed}: {pct:.2f}%")

Pit Bull Mix: 37.15%
Chihuahua Shorthair Mix: 46.99%
Labrador Retriever Mix: 49.45%
German Shepherd Mix: 47.28%
Australian Cattle Dog Mix: 55.23%


In [58]:
color_counts =  cleaned_combo['color'].value_counts()

In [59]:
top_colors = color_counts.head(5)

In [60]:
print(top_colors)

color
Black/White    8270
Black          6673
Brown Tabby    4471
Brown          3598
White          2835
Name: count, dtype: int64


In [61]:
## dictionary with color as key pct as val
percentage_adopted_by_color = {}
for color in top_colors.index:
    total = cleaned_combo[cleaned_combo['color'] == color].shape[0]
    adopted = cleaned_combo[(cleaned_combo['color'] == color) & (cleaned_combo['outcome_type'] =='adoption')].shape[0]
    percentage_adopted_by_color[color] = (adopted / total) * 100

## ******Adoption rates based on color

In [62]:
for color, pct in percentage_adopted_by_color.items():
    print(f"{color}: {pct:.2f}%")

Black/White: 45.20%
Black: 40.34%
Brown Tabby: 41.85%
Brown: 22.07%
White: 37.57%


In [63]:
intake_outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79672 entries, 0 to 79671
Data columns (total 41 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   age_upon_outcome            79672 non-null  object        
 1   animal_id_outcome           79672 non-null  object        
 2   date_of_birth               79672 non-null  object        
 3   outcome_subtype             36348 non-null  object        
 4   outcome_type                79662 non-null  object        
 5   sex_upon_outcome            79671 non-null  object        
 6   age_upon_outcome_(days)     79672 non-null  int64         
 7   age_upon_outcome_(years)    79672 non-null  float64       
 8   age_upon_outcome_age_group  79672 non-null  object        
 9   outcome_datetime            79672 non-null  datetime64[ns]
 10  outcome_month               79672 non-null  int64         
 11  outcome_year                79672 non-null  int64     

In [64]:
fixed_in_shelter = intake_outcomes[intake_outcomes['sex_upon_intake'].str.contains('intact', na=False)].copy()


In [65]:
intake_outcomes['outcome_datetime'] = pd.to_datetime(intake_outcomes['outcome_datetime'])
fixed_in_shelter['month'] = fixed_in_shelter['outcome_datetime'].dt.to_period('M')


In [66]:
fixed_in_shelter['month'] = fixed_in_shelter['outcome_datetime'].dt.to_period('M')

# Count spay/neuter per month
monthly_fixed = fixed_in_shelter.groupby('month').size()
print(monthly_fixed.describe())


count      55.000000
mean      891.290909
std       232.715585
min        40.000000
25%       726.000000
50%       894.000000
75%      1049.000000
max      1496.000000
dtype: float64


## *******Monthly fixed since October 2010

In [67]:
print(monthly_fixed)

month
2013-10     700
2013-11     748
2013-12     924
2014-01     717
2014-02     623
2014-03     830
2014-04     846
2014-05    1082
2014-06    1193
2014-07    1496
2014-08    1250
2014-09    1101
2014-10    1007
2014-11     919
2014-12     887
2015-01     720
2015-02     648
2015-03     643
2015-04     695
2015-05    1035
2015-06    1212
2015-07    1301
2015-08    1107
2015-09     973
2015-10     953
2015-11    1010
2015-12     841
2016-01     785
2016-02     646
2016-03     732
2016-04     672
2016-05     957
2016-06    1076
2016-07    1100
2016-08     976
2016-09    1057
2016-10     850
2016-11     894
2016-12     928
2017-01     783
2017-02     739
2017-03     699
2017-04     768
2017-05     987
2017-06    1091
2017-07    1194
2017-08    1139
2017-09     955
2017-10    1041
2017-11     750
2017-12     755
2018-01     697
2018-02     603
2018-03     646
2018-04      40
Freq: M, dtype: int64
